packages/core/core-flows/src/order/steps/cancel-orders.ts

```diff
@@ -0,0 +1,50 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { IOrderModuleService } from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+type CompleteOrdersStepInput = {
+  orderIds: string[]
+}
+
+export const cancelOrdersStepId = "cancel-orders"
+export const cancelOrdersStep = createStep(
+  cancelOrdersStepId,
+  async (data: CompleteOrdersStepInput, { container }) => {
+    const service = container.resolve<IOrderModuleService>(
+      ModuleRegistrationName.ORDER
+    )
+
+    const orders = await service.list(
+      {
+        id: data.orderIds,
+      },
+      {
+        select: ["id", "status"],
+      }
+    )
```

Comment:
question: Would it be OK to pass the `data.orderIds` directly to `cancel`?

Reply:
I need to query it before to be able to revert on the compensation.

Reply:
but I'm not sure I undertood your question correctly.